salt/states/keystone.py

```diff
@@ -173,11 +182,11 @@ def user_present(name,
             ret['comment'] = 'User "{0}" has been updated'.format(name)
             ret['changes']['Password'] = 'Updated'
         if roles:
-            for tenant_role in roles:
+            for tenant in roles.keys():
```

Comment:
It does* but calling `.keys()`  
- makes the function fail at this point with a useful error msg if a non-dict parameter is passed (instead of a difficult to understand one like I got) and
- clearly states "`roles` is a dict" in this part of the code.

Making `roles` default to an empty dict would also make the `if roles:` a few lines above superfluous but removing all this indention would change a lot of lines.

*) something I just learned today…

Am 21. April 2015 19:09:37 MESZ, schrieb Thomas S Hatch <notifications@github.com>:
>> @@ -173,11 +182,11 @@ def user_present(name,
>>              ret['comment'] = 'User "{0}" has been
>updated'.format(name)
>>              ret['changes']['Password'] = 'Updated'
>>          if roles:
>> -            for tenant_role in roles:
>> +            for tenant in roles.keys():
>
>We should not need to call .keys(), just iterating over the dict should
>return the keys
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/saltstack/salt/pull/22894/files#r28799684